packages/ibm-products/src/global/js/utils/replaceWithOverflow.ts

```diff
@@ -0,0 +1,47 @@
+//
+// Copyright IBM Corp. 2024, 2024
+//
+// This source code is licensed under the Apache-2.0 license found in the
+// LICENSE file in the root directory of this source tree.
+//
+
+import { RefObject } from 'react';
+
+/**
+ * These functions are used to calculate if the content of an area is overflowing it's parent.
+ * Ideally used to replace an element with another if it's overflowing and vice versa.
+ * Check PageHeader.tsx and PageHeaderTitle.js for example usage.
+ */
+
+type Props = {
+  /**
+   * the ref for the element not overflowing
+   */
+  ref: RefObject<HTMLElement>;
+  /**
+   * the ref for the element that is overflowing
+   */
+  overflowRef: RefObject<HTMLElement>;
+};
+
+export const replaceOverflowWidth = ({ ref, overflowRef }: Props): boolean => {
```

Comment:
Stilling thinking this one through, but interested in your thoughts around combining these two utilities and having it default to check `width`, ie something like:
```ts
type Props = {
  ref: RefObject<HTMLElement>;
  overflowRef: RefObject<HTMLElement>;
  mode: 'width' | 'height'
};

export const replaceOverflow = ({ ref, overflowRef, mode = 'width' }) => {
...
}
```

Not necessarily saying `mode` is the best naming, but it would provide a way to merge these utilities so we just had one. I also agree with @amal-k-joy that it might make sense to change the naming, `shouldOverflow`, `willOverflow`, `checkOverflow`?

Reply:
This really starts to hit on some of the issues detailed in https://github.com/carbon-design-system/ibm-products/issues/6274 🎉

Reply:
Ok, yeah that makes sense, a bit easier to use "right out of the box". Lets stick with this approach.